src/client/components/pages/Schedule/__tests__/SchedulePage.test.tsx

```diff
@@ -88,6 +89,110 @@ describe('Schedule Page', function () {
+      });
+    });
+    context('Both is selected from dropdown menu', function () {
+      it('should not fade any course-listing buttons', function () {
+        const { getByLabelText, getAllByText } = renderResult;
+        const degreeProgramDropDown = getByLabelText(/Degree Program/i) as HTMLSelectElement;
+        fireEvent.change(degreeProgramDropDown,
+          { target: { value: DEGREE_PROGRAM.BOTH } });
+        courses.forEach((course) => {
+          const prefixBlocks = getAllByText(course.coursePrefix);
+          const targetBlock = prefixBlocks.find((block) => (
+            within(block.parentElement).queryByText(course.courseNumber)
+          ));
+          const targetCourseListing = within(
+            targetBlock.parentElement
+          ).getByText(course.courseNumber);
+          strictEqual(targetCourseListing.hasAttribute('disabled'), false, `${course.coursePrefix} ${course.courseNumber}-was disabled when it shouldt be`);
+        });
+      });
+    });
+    context('undergraduate is selected from dropdown menu', function () {
+      it('should not fade undergraduate course program', function () {
+        const { getByLabelText, getAllByText } = renderResult;
+        const degreeProgramDropDown = getByLabelText(/Degree Program/i) as HTMLSelectElement;
+        fireEvent.change(degreeProgramDropDown,
+          { target: { value: DEGREE_PROGRAM.UNDERGRADUATE } });
+        courses.forEach((course) => {
+          const prefixBlocks = getAllByText(course.coursePrefix);
+          const targetBlock = prefixBlocks.find((block) => (
+            within(block.parentElement).queryByText(course.courseNumber)
+          ));
+          const targetCourseListing = within(
+            targetBlock.parentElement
+          ).getByText(course.courseNumber);
+          strictEqual(targetCourseListing.hasAttribute('disabled'), !course.isUndergraduate, `${course.coursePrefix} ${course.courseNumber}-was disabled when it shouldn't be`);
+        });
+      });
+    });
+    context('graduate course is selected from dropdown menu', function () {
+      it('should not fade graduate course program', function () {
+        const { getByLabelText, getAllByText } = renderResult;
+        const degreeProgramDropDown = getByLabelText(/Degree Program/i) as HTMLSelectElement;
+        fireEvent.change(degreeProgramDropDown,
+          { target: { value: DEGREE_PROGRAM.GRADUATE } });
+        courses.forEach((course) => {
+          const prefixBlocks = getAllByText(course.coursePrefix);
+          const targetBlock = prefixBlocks.find((block) => (
+            within(block.parentElement).queryByText(course.courseNumber)
+          ));
+          const targetCourseListing = within(
+            targetBlock.parentElement
+          ).getByText(course.courseNumber);
+          strictEqual(targetCourseListing.hasAttribute('disabled'), course.isUndergraduate, `${course.coursePrefix} ${course.courseNumber}-was disabled when it shouldn't be`);
```

Comment:
I would check that the course is not a graduate course rather than checking that the course is undergraduate to future proof this test in case we have more categorizations of courses (beyond undergraduate & graduate) in the future.

Reply:
Hi in the data we have now, the courses only has one prop `isUndergraduate: true` or `isUndergraduate: false` unless we create an array for graduate courses in test. should i loop through the data and create grad course array?

Reply:
Yes I think that would make sense here.